packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -220,19 +220,31 @@ export function useGridDimensions(
       return;
     }
 
-    const height = mainEl.clientHeight || 0;
-    const width = mainEl.clientWidth || 0;
+    const rect = mainEl.getBoundingClientRect();
+    const height = rect.height || 0;
+    const width = rect.width || 0;
```

Comment:
How about using `getComputedStyle()` instead?
```ts
const height = parseFloat(computedStyle.height) || 0;
const width = parseFloat(computedStyle.width) || 0;
```
These values do not include paddings or borders, so additional calculations won't be necessary.